backend/portal/serializers.py

```diff
@@ -10,81 +11,69 @@ class Meta:
-        if club_code not in [
-            x["club"]["code"] for x in get_user_clubs(self.context["request"].user)
-        ]:
+        if not any([x["club"]["code"] == club_code for x in get_user_clubs(user)]):
             raise serializers.ValidationError(
                 detail={"detail": "You do not access to create a Poll under this club."}
             )
+
         # ensuring user cannot create an admin comment upon creation
         validated_data["admin_comment"] = None
-        validated_data["status"] = Poll.STATUS_DRAFT
-
-        # TODO: toggle this off when multiselect functionality is available
-        validated_data["multiselect"] = False
-
-        year = False
-        major = False
-        school = False
-        degree = False
-
-        for population in validated_data["target_populations"]:
-            if population.kind == TargetPopulation.KIND_YEAR:
-                year = True
-            elif population.kind == TargetPopulation.KIND_MAJOR:
-                major = True
-            elif population.kind == TargetPopulation.KIND_SCHOOL:
-                school = True
-            elif population.kind == TargetPopulation.KIND_DEGREE:
-                degree = True
-
-        if not year:
-            validated_data["target_populations"] += list(
-                TargetPopulation.objects.filter(kind=TargetPopulation.KIND_YEAR)
-            )
-        if not major:
-            validated_data["target_populations"] += list(
-                TargetPopulation.objects.filter(kind=TargetPopulation.KIND_MAJOR)
-            )
-        if not school:
-            validated_data["target_populations"] += list(
-                TargetPopulation.objects.filter(kind=TargetPopulation.KIND_SCHOOL)
-            )
-        if not degree:
-            validated_data["target_populations"] += list(
-                TargetPopulation.objects.filter(kind=TargetPopulation.KIND_DEGREE)
+        validated_data["status"] = Content.STATUS_DRAFT
+
+        # auto add all target populations of a kind if not specified
+        auto_add_kind = [
+            kind
+            for kind, _ in TargetPopulation.KIND_OPTIONS
+            if not any(
+                population.kind == kind for population in validated_data["target_populations"]
             )
+        ]
+        validated_data["target_populations"] += TargetPopulation.objects.filter(
+            kind__in=auto_add_kind
+        )
+
+        validated_data["creator"] = user
 
         return super().create(validated_data)
 
     def update(self, instance, validated_data):
-        # if Poll is updated, then approve should be false
+        # if Content is updated, then approve should be false
+
         if not self.context["request"].user.is_superuser:
-            validated_data["status"] = Poll.STATUS_DRAFT
+            validated_data["status"] = Content.STATUS_DRAFT
         return super().update(instance, validated_data)
 
 
+class PollSerializer(ContentSerializer):
```

Comment:
I think its fine, because also this means we don't have to change when we add multiselect. And even if a malicious actor sent us multiselect=True, we need to first approve it and even if we approve it won't do anything cuz the frontend code doesn't use it right now

Reply:
Sounds good